IPython/core/interactiveshell.py

```diff
@@ -453,7 +453,7 @@ def profile(self):
 
 
     # Private interface
-    _post_execute = Instance(dict)
+    _post_execute = Instance(dict, allow_none=True)
```

Comment:
This should actually be a plain Dict. Not sure how it got Instance. Can you remove the `self._post_execute = {}` in `__init__` as well?